tests/test_preprocess.py

```diff
@@ -102,6 +139,55 @@ def test_transform_without_fit(self, preprocessor, raw_field):
         with pytest.raises(ValueError):
             preprocessor.transform(raw_field)
 
+    def test_fit_and_transform_no_climatology(
+        self, preprocessor_no_climatology, raw_field
+    ):
+        preprocessor_no_climatology.fit(raw_field)
+        results = preprocessor_no_climatology.transform(raw_field)
+        expected = preprocess._subtract_trend(
+            raw_field, "linear", preprocessor_no_climatology.trend
+        )
+        assert results == expected
+
+    def test_fit_and_transform_no_detrend(self, preprocessor_no_detrend, raw_field):
+        preprocessor_no_detrend.fit(raw_field)
+        results = preprocessor_no_detrend.transform(raw_field)
+        expected = (
+            raw_field.groupby("time.dayofyear") - preprocessor_no_detrend.climatology
+        )
+        assert results == expected
+
+    def test_fit_and_transform_no_climatology_and_detrend(self, raw_field):
+        prep = preprocess.Preprocessor(
+            rolling_window_size=10,
+            detrend=None,
+            subtract_climatology=False,
+        )
+        prep.fit(raw_field)
+        results = prep.transform(raw_field)
+
+        assert results == raw_field
+
     def test_fit_transform(self, preprocessor, raw_field):
         preprocessed_data = preprocessor.fit_transform(raw_field)
         assert preprocessed_data is not None
+
+    def test_trend_property_not_fit(self, preprocessor):
+        with pytest.raises(ValueError, match="The preprocessor has to be fit"):
+            preprocessor.trend
+
+    def test_trend_property_no_detrend(self, preprocessor_no_detrend, raw_field):
+        preprocessor_no_detrend.fit(raw_field)
+        with pytest.raises(ValueError, match="Detrending is set to `None`"):
+            preprocessor_no_detrend.trend
+
+    def test_climatology_property_not_fit(self, preprocessor):
+        with pytest.raises(ValueError, match="The preprocessor has to be fit"):
+            preprocessor.climatology
+
+    def test_trend_property_no_climatology(
+        self, preprocessor_no_climatology, raw_field
+    ):
+        preprocessor_no_climatology.fit(raw_field)
```

Comment:
```suggestion
```
This step should not be needed right? If `subtract_climatology` is `False` then no matter if the preprocessor is fit or not, it can't give the climatology :)

Reply:
Hmmm....I just noticed that I forgot to move "no climatology" error in the front of the "not fit" error. Yes, afterward this line is not needed.